pkg/scale/encode_test.go

```diff
@@ -1096,6 +1114,24 @@ func Test_encodeState_encodeArray(t *testing.T) {
 	}
 }
 
+func Test_encodeState_encodeMap(t *testing.T) {
+	for _, tt := range mapTests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
you should pin the `tt` variable to the loop context otherwise it can cause inconsistency
```suggestion
	for _, tt := range mapTests {
	        tt := tt
		t.Run(tt.name, func(t *testing.T) {
```